apps/api/src/app/shared/shared.module.ts

```diff
@@ -106,6 +107,7 @@ const distributedLockService = {
 const PROVIDERS = [
   cacheService,
   distributedLockService,
+  inMemoryProviderService,
```

Comment:
Is there any reason why we are creating 2 instances of the `InMemoryProviderService`?
we can inject it like this: 
```
{
    provide: CacheService,
    useFactory: (inMemoryProviderService: InMemoryProviderService) => {
      return new CacheService(inMemoryProviderService);
    },
    inject: [InMemoryProviderService],
  },
```

Reply:
We are configuring the provider with auto pipelining for the CacheService and without for the DistributedLockService. That is one of the reasons. 🙁